exporter/exporterhelper/obsexporter.go

```diff
@@ -252,3 +273,43 @@ func toNumItems(numExportedItems int, err error) (int64, int64) {
 	}
 	return int64(numExportedItems), 0
 }
+
+func (or *ObsReport) recordEnqueueFailure(ctx context.Context, dataType component.DataType, failed int64) {
+	if or.useOtelForMetrics {
+		or.recordEnqueueFailureWithOtel(ctx, dataType, failed)
+	} else {
+		or.recordEnqueueFailureWithOC(ctx, dataType, failed)
+	}
+}
+
+func (or *ObsReport) recordEnqueueFailureWithOC(ctx context.Context, dataType component.DataType, failed int64) {
+	var failedMeasure *stats.Int64Measure
+	switch dataType {
+	case component.DataTypeTraces:
+		failedMeasure = obsmetrics.ExporterFailedToEnqueueSpans
+	case component.DataTypeMetrics:
+		failedMeasure = obsmetrics.ExporterFailedToEnqueueMetricPoints
+	case component.DataTypeLogs:
+		failedMeasure = obsmetrics.ExporterFailedToEnqueueLogRecords
+	}
+	if failed > 0 {
+		_ = stats.RecordWithTags(
+			ctx,
+			or.mutators,
+			failedMeasure.M(failed))
+	}
+}
+
+func (or *ObsReport) recordEnqueueFailureWithOtel(ctx context.Context, dataType component.DataType, failed int64) {
```

Comment:
I'm wondering if this new otel metric can be updated externally through the API? This package allows an external package to call `exporterhelper.ObsReport.StartTracesOp`  and `exporterhelper.ObsReport.EndTracesOp`. And `EndTracesOp` will call `recordMetrics` which will record `sent` and `failedToSend` - i.e. can the API also update `failedToEnqueue`?

Reply:
IMO StartTracesOp and EndTracesOp should be able to catch the error from the queue sender, but the individual sub-components are not connected in the right order for this. I do not want to replicate existing functionality, for the record, @moh-osman3, I think we should focus on a new spec: https://github.com/open-telemetry/semantic-conventions/pull/184

Reply:
> will call recordMetrics which will record sent and failedToSend - i.e. can the API also update failedToEnqueue?

I originally considered going the route of recordMetrics updating all 3 metrics, but sent/failedToSend are just opposites of each other, which makes it easy to record together. I suppose recording enqueue failed could be done by checking the error type in `End*Op`. Whether we decide to do this or not (based on @jmacd's comment) I would do this as a follow up PR, as that's a change in behaviour. This PR is really just about ensuring that the metric is emitted in both the OC and the Otel case :)

Would be happy to open an issue to follow up @moh-osman3

Reply:
That makes sense thanks @codeboten @jmacd! I was mainly asking because I was writing some code in the otel-arrow repository and realized I wouldn't have access to that metric if I tried to access obsreport externally. But as Josh pointed out we might just do a simpler instrumentation from scratch instead of depending on that obsreport package